### PR TITLE
🧩 Refactor 🧩

### PR DESCRIPTION
**Pull request by sam-at-pieces**
Quality of life fixes:
- standardizes package structure so it is inline with pieces_os_client
- adds dependency and build manager with Poetry
- adds unit tests
- adds github action CI/CD workflow for building the library

### CODE REVIEW

**Comment by sam-at-pieces** [comment]
@hal-8999-alpha ?

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  pyperclip = "1.8.2"
+  requests = "2.31.0"
+  typing-extensions = "4.9.0"
+  urllib3 = "2.0.7"
+  websocket-client = "1.7.0"
+  openapi_client = "*"
+  pieces_os_client = "*"
+
+[tool.poetry.dev-dependencies]
+  pytest = "^8.0.0"
```
[the remark this answers is not in the document]
Comment (mark-at-pieces) [reply pyproject.toml:23]: super solid, so basically how we are pulling in pieces_os_client now says that it is pointing to a local folder?

File: src/pieces/__initi__.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Importing the WebSocketManager class
```
Comment (sam-at-pieces) [inline src/pieces/__initi__.py:1]: I've exposed nearly every funciton in the library here, prob want to be more selective

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  aenum = "3.1.15"
+  certifi = "2023.11.17"
+  charset-normalizer = "3.3.2"
+  pydantic = "1.10.13"
+  pyperclip = "1.8.2"
+  requests = "2.31.0"
+  typing-extensions = "4.9.0"
+  urllib3 = "2.0.7"
+  websocket-client = "1.7.0"
+  openapi_client = "*"
```
Comment (jordan-pieces) [inline pyproject.toml:19]: this also should likely be removable as i think it is coming from the old package name

**Comment by sam-at-pieces** [comment]
@hal-8999-alpha how are we doing on this, good to merge this evening?

**Review by mark-at-pieces — APPROVED** [review]
LGTM

**Comment by hal-8999-alpha** [comment]
> @hal-8999-alpha how are we doing on this, good to merge this evening?

Good stuff! Successfully ran it locally and in code spaces with a one tweak and a suggestion: 

1. In requirments.txt change pieces-os-client to 1.2.3
2. Add instructions to cd into dist and run pip install pieces-0.1-py3-none-any.whl after building via poetry or make it all part of the installation process

**Pull request merged**